### PR TITLE
Login / Register / Spinner components including Unit tests

### PR DESCRIPTION
**Pull request by valerieyang00**
Main changes made:

Login.tsx along with Login.test.tsx
Register.tsx along with Register.test.tsx
Spinner.tsx along with Spinner.test.tsx
New hook file for LoginUser
Geolocation added to LandingMap but still need some work on this component

### CODE REVIEW

File: src/components/pages/Login.test.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  jest.spyOn(Object.getPrototypeOf(window.localStorage), 'setItem')
+  Object.setPrototypeOf(window.localStorage.setItem, jest.fn())
+
+  test('renders login form without crashing', () => {
+    render(
+      <QueryClientProvider client={new QueryClient()}>
+        <Router>
+          <Login />
+        </Router>
+      </QueryClientProvider>,
```
Comment (moyara) [inline src/components/pages/Login.test.tsx]:
```
<QueryClientProvider client={new QueryClient()}>
  <Router>
     ....
```

There is repeative part used for every render. We can create something like TestProvider which contains the common part such as those QueryClientProvider and Router.

**Review by moyara — COMMENTED** [review]
looks good. looks like you are still working for LandingMap? probably we can exclude the file and create a separate PR with tests.

File: src/components/pages/Login.test.tsx (hunk 1 above)
Comment (valerieyang00) [reply src/components/pages/Login.test.tsx]: Just pushed up the changes. Is this ok?

**Review by moyara — APPROVED** [review]

**Comment by moyara** [comment]
Nice work!

**Pull request merged**